12.0 refactor css 2

ashleyguevarra opened the pull request.

ashleyguevarra committed:
[UPD] website_accorderie: Refactorisation des fichiers CSS,SCSS,XML

-------------------------------------------------------------------
-> Créez une structure pour les fichiers CSS par section de page:
	Il divise le code par section, par exemple le code de la page
	d'accueil dans une catégorie, il met le code de la page
	communautaire dans une autre catégorie, etc.

-> Classez les propriétés par ordre alphabétique:
	Utilisant > Sort Lines Ascending dans le search.

-> Polissage de tous les fichiers:
	En espaçant les différentes classes et propriétés
	pour que ce soit plus propre et plus lisible.

-> Correctif des bogues:
	-> Correction du bug des couleurs sur la bannière du profil,
	   la page de mon profil, mon activité : mise en place avec
	   le $text-gamma.

	-> Dans la page de profil le nombre d'heures quand on
	   réduit la page il n'est pas au bon endroit, donc ceci a
	   diminué la taille du texte de l'heure.

ashleyguevarra committed:
[ADD] website_accorderie: Flexbox layout

-------------------------------------------------------------------
-> La disposition en grille a été remplacée par une disposition flexbox
   dans la page de mon activité:
    La disposition flexbox est idéale pour les mises en page
    unidimensionnelles ou à une seule ligne, où il suffit d'espacer
    un groupe d'éléments d'une certaine manière.

-> Ceci utilise l'imbrication Flexbox (Nesting Flexbox):
    Cette méthode permet de créer des mises en page bidimensionnelles
    en imbriquant un conteneur flexible à l'intérieur d'un autre conteneur.

mathben commented on .gitignore:
```diff
@@ -40,8 +40,7 @@ pip-delete-this-directory.txt
 .tox/
-.nox/
-.coverage
+.nox/ .coverage
```
C'est une erreur, à revert

mathben commented on website_accorderie/data/ir_ui_view.xml:
```diff
@@ -3066,38 +3066,41 @@
-                                    <div class="bilan_container">
-                                        <a href="/monactivite/echanges" class="bilan bilan-1 btn" style="background-image: url(&quot;/website_accorderie/static/src/img/abstract-blue-and-green-fluid-shape-modern-background-vector.jpg&quot;)">
+                                <div class="bilan_container">
+                                    <nav class="bilan_left">
```
Ça semble un bon choix le nav. Pourquoi enlever section?

mathben commented on website_accorderie/data/ir_ui_view.xml:
```diff
@@ -3066,38 +3066,41 @@
                                         </a>
-                                        <a href="/monactivite/echanges" class="bilan bilan-2 btn" style="background-image: url(&quot;https://i.pinimg.com/originals/60/43/bf/6043bf9787ade7d5b2e746ad0960cd5a.jpg&quot;)">
+                                    </nav>
+                                    <section class="bilan_main">
```
Ici c'est encore une section, plus haut, C'est un nav.

mathben commented on website_accorderie/data/ir_ui_view.xml:
```diff
@@ -3066,38 +3066,41 @@
-                                        <a href="/monactivite/echanges" class="bilan bilan-2 btn" style="background-image: url(&quot;https://i.pinimg.com/originals/60/43/bf/6043bf9787ade7d5b2e746ad0960cd5a.jpg&quot;)">
+                                    </nav>
+                                    <section class="bilan_main">
+                                        <a href="/monactivite/echanges" class="bilan_main bilan-2 btn" style="background-image: url(&quot;https://i.pinimg.com/originals/60/43/bf/6043bf9787ade7d5b2e746ad0960cd5a.jpg&quot;)">
```
Il faudrait enlever tous liens externes d'image, tout doit être local. Un oublie. Ça augmente les performances d'avoir tout au même endroit.

mathben commented on website_accorderie/data/ir_ui_view.xml:
```diff
@@ -3066,38 +3066,41 @@
                                             <span class="nbr">{{personal.nb_echange_en_cours}}</span>
                                             <span class="txt">échange en cours</span>
                                         </a>
-<!--                                        <span class="debug_unimplemented">-->
```
Il faut garder les debug_unimplemented. C'est une fonctionnalité future à implémenté. S'il a été commenté, c'est à cause d'un problème d'affichage. Ce qui ne fonctionne pas doit être caché. ET c'est pourquoi tu as implémenté les flexbox, pour faciliter la gestion d'ajout/enlever les boutons.

mathben commented on website_accorderie/static/src/scss/communaute/membres.scss:
```diff
@@ -0,0 +1,88 @@
+  }
+  
+  .item_foot:after {
+    content: "";
```
C'est intéressant ça, je ne savais pas que ça existait. J'ai un problème sur le formatage du code avec prettier-xml et je crois que ça pourrait résoudre le problème, c'est qu'il y a des espaces d'ajouter dans le rendu lorsqu'on format tout (exemple <i icon>).

mathben commented on website_accorderie/static/src/scss/explorer/categories.scss:
```diff
@@ -0,0 +1,50 @@
+    Offres catégories
+*/
+
+.s_offres {
```
Problème de formatage, il faut appeler `make format_accorderie` à la racine de ERPLibre.

mathben commented on website_accorderie/static/src/scss/monactivite/bilan.scss:
```diff
@@ -0,0 +1,162 @@
+  white-space: normal;
+}
+
+.bilan-1 {
```
Est-ce qu'on pourrait mettre un nom au lieu d'un chiffre? S'il y a plus de 5 bilans, on fait quoi?

mathben replied:
Par exemple, on peut dire pairs, impairs, début, centre, fin. Donc trouver un terme significatif selon le contexte lié à une caractéristique.

mathben commented on website_accorderie/static/src/scss/monactivite/echange_a_venir.scss:
```diff
@@ -0,0 +1,166 @@
+      }
+    }
+  
+    .information-1 {
```
même chose ici, peut-on trouver une caractéristique pour définir le chiffre?

mathben commented on website_accorderie/static/src/scss/monactivite/echange_a_venir.scss:
```diff
@@ -0,0 +1,166 @@
+    }
+
+    .information-4 {
+      grid-area: 3 / 2 / 4 / 3;
```
On utilise encore le grid-area?

mathben reviewed (changes requested):
Beau travail, belle avancement, ça va faire du bien cette refactorisation. Il reste encore des petits détails à régler. On pourrait merger pour débloquer et revenir plus tards aux points mentionnés.
Il faut formater avant de merger. J'attends ton retour.

ashleyguevarra committed:
[UPD] website_accorderie: Refactorisation sur ORE

-------------------------------------------------------------------
[Fix des bogues]:

[Login]
 -> Changer le titre pour : Bienvenue sur l’espace membre de ORE.
 -> Enlever l'image du vélo et mettre une nouvelle photo.

[Offrir un service]
 -> Formulaire : Changer le background et enlever le bouton : Remplacer la photo.
 -> Valider un échange - Oui - Clicker sur un offre : Changer le border du box pour
    la couleur orange et la couleur jaune du background du box pour la couleur blanc.
    Aussi, pour le montant total, changer la couleur du background pour rose.
 -> Si on hover sur un box, mettre le background gris et si on click sa devient toute orange.

[Pages]
 -> Page Communauté, Page Mon activité, Page Mon compte: Changer le background et mettre le
    texte mauve au lieu d’orange.
 -> Page Mon profil: Changer les couleurs des flexbox pour Mauve, Orange. (Couleur priorité: Mauve,
    Orange, Jaune, Rose).
 -> Page Mon profil/Ma présentation: Changer les textes du Mes centres d'intérêt pour mauve et boîte
    gris. (Communauté/Membres 	[Ligne 58] Changer le background et la couleur des textes des buttons.
 -> Page Mon profil/Mes favoris: Mettre le coeur en rose au lieu d’orange.
 -> Dans chaque navbar : Changer le background.
 -> Page Communauté et Mon Activité: Changer la position du nom et de l’âge.
 -> Changer la couleur du texte du navbar “Publier” et le nom de la personne en blanc.

[General]
 -> Navbar Jaune : Mettre le gris PARTOUT.
 -> Navbar Gris (2ième) : Enlever PARTOUT.
 -> Changer tous les textes orange/mauve en blanc/mauve.

[Explorer]
 -> Changer le background. [Ligne 1481]
